src/cmd/pay.rs

```diff
@@ -111,40 +129,55 @@ impl Cmd {
 
     fn collect_payments(&self) -> Result<Vec<Payment>> {
         match &self {
-            Self::One(one) => Ok(vec![Payment {
-                payee: one.payee.address.to_vec(),
-                // we safely create u64 from the amount of type Token
-                // only because each token_type has the same amount of decimals
-                amount: u64::from(one.payee.amount),
-                memo: u64::from(&one.payee.memo),
-                max: false,
-                token_type: match one.payee.token {
-                    TokenInput::Hnt => BlockchainTokenTypeV1::Hnt.into(),
-                    TokenInput::Hst => BlockchainTokenTypeV1::Hst.into(),
-                    TokenInput::Iot => BlockchainTokenTypeV1::Iot.into(),
-                    TokenInput::Mobile => BlockchainTokenTypeV1::Mobile.into(),
-                },
-            }]),
+            Self::One(one) => {
+                let amount_is_max = one.amount.eq("max");
+                Ok(vec![Payment {
+                    payee: one.address.to_vec(),
+                    // we safely create u64 from the amount of type Token
+                    // only because each token_type has the same amount of decimals
+                    amount: if amount_is_max {
+                        0
+                    } else {
+                        u64::from(Token::from_str(&one.amount)?)
+                    },
+                    memo: u64::from(&one.memo),
+                    max: amount_is_max,
+                    token_type: match one.token {
+                        TokenInput::Hnt => BlockchainTokenTypeV1::Hnt.into(),
+                        TokenInput::Hst => BlockchainTokenTypeV1::Hst.into(),
+                        TokenInput::Iot => BlockchainTokenTypeV1::Iot.into(),
+                        TokenInput::Mobile => BlockchainTokenTypeV1::Mobile.into(),
+                    },
+                }])
+            }
             Self::Multi(multi) => {
                 let file = std::fs::File::open(multi.path.clone())?;
                 let payees: Vec<Payee> = serde_json::from_reader(file)?;
-                let payments = payees
-                    .iter()
-                    .map(|p| Payment {
+                let mut payments: Vec<Payment> = Vec::new();
+                for p in payees {
```

Comment:
I changed this from `map` to `for` so that I could do the sanity check on max and amount and bail if needed. It's less funtional/fluent/iterator-y, but I think the sanity check will help users debug their own JSON without submitting it.